skema/skema-rs/mathml/src/parsers/decapodes_serialization.rs

```diff
@@ -0,0 +1,441 @@
+                table_counts.variable_count += 1;
+                let tgt_idx = table_counts.variable_count;
+
+                let unary = UnaryOperator {
+                    src: to_decapodes_serialization(&rest[0], tables, table_counts),
+                    tgt: tgt_idx,
+                    op1: "Abs".to_string(),
+                };
+                tables.unary_operators.push(unary.clone());
+                tgt_idx
+            }
+            Operator::Add => {
+                table_counts.sum_count += 1;
+                let temp_sum = format!("sum_{}", table_counts.sum_count);
+                let temp_variable = Variable {
+                    r#type: Type::infer,
+                    name: temp_sum,
+                };
+                tables.variables.push(temp_variable.clone());
+                table_counts.variable_count += 1;
+                let summing = Sum {
+                    sum: table_counts.variable_count,
+                };
+                tables.sum_op.push(summing.clone());
+                let tgt_idx = table_counts.variable_count;
+                let sum_tgt_idx = table_counts.sum_count;
+
+                for r in rest.iter() {
+                    let summands = Summation {
+                        summand: to_decapodes_serialization(r, tables, table_counts),
+                        summation: sum_tgt_idx,
+                    };
+                    tables.summand_op.push(summands.clone());
+                }
+                tgt_idx
+            }
+            Operator::Power => {
+                table_counts.operation_count += 1;
+                let temp_str = format!("•{}", table_counts.operation_count);
+                let temp_variable = Variable {
+                    r#type: Type::infer,
+                    name: temp_str,
+                };
+                tables.variables.push(temp_variable.clone());
+                table_counts.variable_count += 1;
+                let tgt_idx = table_counts.variable_count;
+
+                let projection = ProjectionOperator {
+                    proj1: to_decapodes_serialization(&rest[0], tables, table_counts),
+                    proj2: to_decapodes_serialization(&rest[1], tables, table_counts),
+                    res: tgt_idx,
+                    op2: "^".to_string(),
+                };
+                tables.projection_operators.push(projection.clone());
+                tgt_idx
+            }
+            Operator::Subtract => {
+                table_counts.operation_count += 1;
+                let temp_str = format!("•{}", table_counts.operation_count);
+                let temp_variable = Variable {
+                    r#type: Type::infer,
+                    name: temp_str,
+                };
+                tables.variables.push(temp_variable.clone());
+                table_counts.variable_count += 1;
+                let tgt_idx = table_counts.variable_count;
+                let projection = ProjectionOperator {
```

Comment:
You might want to add a check here on if the subtraction id unary or not and have two different flows depending. If it is a urnary subtraction the `rest[1]` call will be out of bounds and make it crash, I suspect.